### PR TITLE
Feature/performance tests

### PR DESCRIPTION
**Pull request by beseven**

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

File: perf-tests/http.jmx [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+          </value>
+        </objProp>
+        <stringProp name="filename"></stringProp>
+      </ResultCollector>
+      <hashTree/>
+      <ConfigTestElement guiclass="HttpDefaultsGui" testclass="ConfigTestElement" testname="HTTP Request Defaults" enabled="true">
+        <elementProp name="HTTPsampler.Arguments" elementType="Arguments" guiclass="HTTPArgumentsPanel" testclass="Arguments" testname="User Defined Variables" enabled="true">
+          <collectionProp name="Arguments.arguments"/>
+        </elementProp>
+        <stringProp name="HTTPSampler.domain">c2s-g-finder.dev.c2s.nhschoices.net</stringProp>
```
Comment (beseven) [inline perf-tests/http.jmx]: @adamgraves-choices  Argh, making the typo change now!

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

File: perf-tests/http.jmx [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        <stringProp name="ThreadGroup.duration">600</stringProp>
+        <stringProp name="ThreadGroup.delay">0</stringProp>
+      </ThreadGroup>
+      <hashTree>
+        <HTTPSamplerProxy guiclass="HttpTestSampleGui" testclass="HTTPSamplerProxy" testname="/book-a-gp-appointment/" enabled="true">
+          <elementProp name="HTTPsampler.Arguments" elementType="Arguments" guiclass="HTTPArgumentsPanel" testclass="Arguments" enabled="true">
+            <collectionProp name="Arguments.arguments"/>
+          </elementProp>
+          <stringProp name="HTTPSampler.domain"></stringProp>
+          <stringProp name="HTTPSampler.port"></stringProp>
```
Comment (adamgraves-choices) [inline perf-tests/http.jmx]: Port needs to be the container port, so 3000

File: perf-tests/http.jmx [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+          <stringProp name="HTTPSampler.connect_timeout"></stringProp>
+          <stringProp name="HTTPSampler.response_timeout"></stringProp>
+          <stringProp name="HTTPSampler.protocol">https</stringProp>
```
Comment (adamgraves-choices) [inline perf-tests/http.jmx]: The container serves HTTP so needs to be `http`

File: perf-tests/http.jmx [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+                <stringProp name="Header.name">User-Agent</stringProp>
+                <stringProp name="Header.value">Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36</stringProp>
+              </elementProp>
+              <elementProp name="Accept" elementType="Header">
+                <stringProp name="Header.name">Accept</stringProp>
+                <stringProp name="Header.value">text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8</stringProp>
+              </elementProp>
+              <elementProp name="Referer" elementType="Header">
+                <stringProp name="Header.name">Referer</stringProp>
+                <stringProp name="Header.value">https://c2s-gp-finder.dev.c2s.nhschoices.net/book-a-gp-appointment/results?search=abbey</stringProp>
```
Comment (adamgraves-choices) [inline perf-tests/http.jmx]: needs to be the service name from `docker-compose`

File: perf-tests/http.jmx [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+          </value>
+        </objProp>
+        <stringProp name="filename"></stringProp>
+      </ResultCollector>
+      <hashTree/>
+      <ConfigTestElement guiclass="HttpDefaultsGui" testclass="ConfigTestElement" testname="HTTP Request Defaults" enabled="true">
+        <elementProp name="HTTPsampler.Arguments" elementType="Arguments" guiclass="HTTPArgumentsPanel" testclass="Arguments" testname="User Defined Variables" enabled="true">
+          <collectionProp name="Arguments.arguments"/>
+        </elementProp>
+        <stringProp name="HTTPSampler.domain">c2s-gp-finder.dev.c2s.nhschoices.net</stringProp>
```
Comment (adamgraves-choices) [inline perf-tests/http.jmx]: needs to be the service name from `docker-compose`

File: perf-tests/http.jmx [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+                <stringProp name="Header.name">User-Agent</stringProp>
+                <stringProp name="Header.value">Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36</stringProp>
+              </elementProp>
+              <elementProp name="Accept" elementType="Header">
+                <stringProp name="Header.name">Accept</stringProp>
+                <stringProp name="Header.value">text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8</stringProp>
+              </elementProp>
+              <elementProp name="Referer" elementType="Header">
+                <stringProp name="Header.name">Referer</stringProp>
+                <stringProp name="Header.value">https://c2s-gp-finder.dev.c2s.nhschoices.net/book-a-gp-appointment/</stringProp>
```
Comment (adamgraves-choices) [inline perf-tests/http.jmx]: needs to be the service name from `docker-compose`

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:warning: deployment of 34 for nhsuk/gp-finder failed

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

**Comment by adamgraves-choices** [comment]
@beseven the VOLUME /code/perf-tests line isn't in the Dockerfile anymore, I assume it's been flattened on  a rebase. That's what's causing the issues.

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-34.dev.c2s.nhschoices.net](http://gp-finder-pr-34.dev.c2s.nhschoices.net)

**Pull request merged**